src/ast.rs

```diff
@@ -356,3 +375,32 @@ pub fn forall<const N: usize>(f: impl FnOnce([Var; N]) -> Rule) -> Rule {
 pub fn exists<const N: usize>(f: impl FnOnce([Var; N]) -> Query) -> Query {
     quantify(f)
 }
+
+#[cfg(test)]
+mod test {
```

Comment:
Thanks for adding these!

Reply:
I also added one for get_or_insert, since I touched it.